pkg/fakedoc/generator.go

```diff
@@ -315,3 +321,30 @@ func (gen *Generator) loremIpsum(minlength, maxlength int, unit LoremUnit) strin
 		return lorem.Words(length)
 	}
 }
+
+func (gen *Generator) book(minlength, maxlength int, path string) (string, error) {
+	if minlength < 0 {
+		minlength = 0
+	}
+	if maxlength < 0 {
+		// FIXME: make bound on maximum length configurable
+		maxlength = minlength + 10
+	}
+
+	length := minlength + gen.Rand.IntN(maxlength-minlength)
+	content, ok := gen.FileCache[path]
+	if !ok {
+		file, err := os.Open(path)
+		if err != nil {
+			return "", err
+		}
+		defer file.Close()
+		content, err = io.ReadAll(file)
+		if err != nil {
+			return "", err
+		}
+		gen.FileCache[path] = content
+	}
+
+	return string(content[:length]), nil
```

Comment:
We should guarantee that this a valid UTF-8 string.

Reply:
An error is now returned if the string is not correct UTF-8. The trimming no longer occurs on the byte boundary, which may cause issues with non-ASCII input.